napari/_vispy/_tests/test_utils.py

```diff
@@ -1,19 +1,43 @@
 import numpy as np
+import pytest
 from vispy.util.quaternion import Quaternion
 
 from napari._vispy.quaternion import quaternion2euler
 
 
-def test_quaternion2euler(qtbot):
+@pytest.mark.parametrize('angles', [[12, 53, 92], [180, -90, 0], [16, 90, 0]])
+def test_quaternion2euler(angles):
     """Test quaternion to euler angle conversion."""
-    # Test roundtrip degrees
-    angles = (12, 53, 92)
+
+    # Test for degrees
     q = Quaternion.create_from_euler_angles(*angles, degrees=True)
     ea = quaternion2euler(q, degrees=True)
-    np.testing.assert_allclose(ea, angles)
+    q_p = Quaternion.create_from_euler_angles(*ea, degrees=True)
+
+    # We now compare the corresponding quaternions ; they should be equals or opposites (as they're already unit ones)
+    q_values = np.array([q.w, q.x, q.y, q.z])
+    q_p_values = np.array([q_p.w, q_p.x, q_p.y, q_p.z])
+
+    nn_zero_ind = np.argmax((q_values != 0) & (q_p_values != 0))
+
+    q_values *= np.sign(q_values[nn_zero_ind])
+    q_p_values *= np.sign(q_p_values[nn_zero_ind])
+
+    np.testing.assert_allclose(q_values, q_p_values)
 
-    # Test roundtrip radians
-    angles = (0.1, -0.2, 1.2)
-    q = Quaternion.create_from_euler_angles(*angles)
+    # Test for radians
```

Comment:
Let's remove this code block and make this a doubly parametrized test as the only thing different here is degrees `True`/ `False`. You can do that by passing a list of 2-tuples `(angle, degrees)` and then pass in 6 values instead of three, see https://docs.pytest.org/en/stable/parametrize.html